### PR TITLE
[READY]Makes /client/s a child of /datums/

### PR DESCRIPTION
**Pull request by MrsTonedOne**
:cl:
experiment: Blessed our beloved code with dark magic.
/:cl:

@tgstation/commit-access 
@hornygranny 
@Fox-McCloud 
http://www.byond.com/forum/?post=2145003

### CODE REVIEW

**Comment by Jordie0608** [comment]
mild

**Comment by Iamgoofball** [comment]
so what do we get from this?

**Comment by MrStonedOne** [comment]
/client/s are now a child of /datum/

That's what we get from this.

Like, I shouldn't have to spell it out for you.

**Comment by Iamgoofball** [comment]
i know

do we get any perf benefits

**Comment by MrStonedOne** [comment]
no
This is about object inheritance excessively

right now if you define a proc on datums, that proc will exist on everything, from icons to images to atoms to mobs to objs to turfs to areas.

Everything is in one way or another, a child of datums,

atom is a child of datums, movable is a child of atoms, mobs are a child of mobable, etc

even images are a child of datums

icons are a child of datums

but not clients. clients aren't, they just ask like they are.

so while i can do `istype(icon, /datum)` and have it return true, i can't do `istype(client, /datum)`, it will always return false, so if i make a proc, that i want to work with any kind of things, like say VV, i have to do this bullshit:

`if (istype(thing, /datum) || istype(thing, /client)`

now i won't have to do that,

**Comment by Jordie0608** [comment]
Is there a reason why clients aren't a child of datums natively?

**Comment by MrStonedOne** [comment]
lummox said he doesn't know when i asked, but he didn't foresee any issues with doing this.

There are a few bugs i have to work out still, some things treat clients differently, but only after they failed a istype() for datum, so i'd have to make it istype for client before it istypes for datum.

**Comment by FlavoredCactus** [comment]
Guess dantom took the knowledge to the grave

**Comment by phil235** [comment]
>so if i make a proc, that i want to work with any kind of things, like say VV, i have to do this bullshit:
if (istype(thing, /datum) || istype(thing, /client))

.
 
>some things treat clients differently, but only after they failed a istype() for datum, so i'd have to make it istype for client before it istypes for datum.

So let's hope the instances of the first case don't outnumber the second case

**Comment by AnturK** [comment]
We don't really have many /datum procs do we ? Mostly VV and i think tgui helpers were generic.

**Comment by MrStonedOne** [comment]
Test merged on sybil

**Comment by KorPhaeron** [comment]
Don't think this needs a changelog

**Comment by MrStonedOne** [comment]
Don't care, it's getting on.

**Comment by Razharas** [comment]
I remember i made a proc that checks for the thing to be either a datum or a client
Guess it can be removed now

**Comment by MrStonedOne** [comment]
where is it, i don't remember seeing it searching for ,/client or , /client

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -966,12 +966,9 @@ var/list/WALLITEMS_INVERSE = list(
 	tY = max(1, min(origin.y + 7 - tY, world.maxy))
 	return locate(tX, tY, tZ)
 
-/proc/IsValidSrc(A)
```
Comment (Razharas) [inline code/__HELPERS/unsorted.dm]: Yep thats the proc i added
Dont even remember what for but eh

**Comment by Cruix** [comment]
But how big does the pentagram have to be?

**Pull request merged**